autogen.sh

```diff
@@ -775,6 +775,13 @@ echo_n "Building ROMIO glue code... "
 ( cd src/glue/romio && chmod a+x ./all_romio_symbols && ./all_romio_symbols ../../mpi/romio/include/mpio.h.in )
 echo "done"
 
+########################################################################
+## Building C interfaces
+########################################################################
+
+echo_n "generating MPI C functions..."
+$PYTHON maint/gen_binding_c.py
```

Comment:
I was curious what would happen if I ran autogen at this commit, and I got an error that the `src/binding/c` directory didn't exist. I think we should consider merging this commit with the Makefile addition, and the actual replacement of pt2pt function with the autogenerated ones in order to consistently be able to build within the PR.

Reply:
Ok, will do.

Reply:
> I agree with @wesbland that some of the code here is probably better suited to autogeneration. But note that all the user argument checking is autogenerated, which is a big win IMO.

> I am not opposed to the concept of the config file. We have this type of capability in buildiface scripts, too. Perhaps more aggressive generation doesn't need to happen in this first pass.

There will be a lot of renaming and consistency refactoring to generate all the body of routines. It is fairly straight forward though. The mechanism is already there. For example, see https://github.com/pmodels/mpich/pull/4846/commits/cd92536769245af73b89b1c62d2c9e62929ab237#r533692730. Yeah, since it is not critical, I would prefer do it separately.